### PR TITLE
Add hygiene information to SourceAnalyzer

### PR DESCRIPTION
**Pull request by edwin0cheng**
This should fix https://github.com/rust-analyzer/rust-analyzer/pull/2392#issuecomment-557964686

### CODE REVIEW

File: crates/ra_ide_api/src/expand_macro.rs [hunk 1]
```diff
@@ -269,4 +269,27 @@ fn some_thing() -> u32 {
...
+        }        
+
+        fn main() {        
+            let res = fo<|>o!();
+        }
+        "#,
+        );
+
+        assert_eq!(res.name, "foo");
+        assert_snapshot!(res.expansion, @r###"0"###);
```
Comment (bjorn3) [inline crates/ra_ide_api/src/expand_macro.rs]: Could you also test invoking `foo` from a crate other than where it is defined?

File: crates/ra_hir/src/source_binder.rs [hunk 2]
```diff
@@ -441,12 +442,14 @@ impl SourceAnalyzer {
...
     ) -> Option<Expansion> {
-        let def = self.resolve_macro_call(db, macro_call.value)?.id;
+        let def = self.resolve_macro_call(db, macro_call)?.id;
         let ast_id = AstId::new(
             macro_call.file_id,
             db.ast_id_map(macro_call.file_id).ast_id(macro_call.value),
         );
         let macro_call_loc = MacroCallLoc { def, ast_id };
+        let kind = to_macro_file_kind(macro_call.value);
+        dbg!(kind);
```
Comment (matklad) [inline crates/ra_hir/src/source_binder.rs]: stray `dbg!`

File: crates/ra_ide_api/src/expand_macro.rs (hunk 1 above)
Comment (matklad) [reply crates/ra_ide_api/src/expand_macro.rs]: I don't think this is really necessary: there should be other test that checks that `$crate` is handled correctly, if the hygiene is present. But this wouldn't hurt either!

**Comment by edwin0cheng** [comment]
bors r=matklad

**Pull request merged**